src/main/java/com/powsybl/openloadflow/ac/equations/AcEquationSystem.java

```diff
@@ -45,6 +45,9 @@ private static void createBusEquations(LfNetwork network, VariableSet<AcVariable
                 createDiscreteVoltageControlEquation(bus, variableSet, equationSystem);
             }
             Equation<AcVariableType, AcEquationType> v = equationSystem.createEquation(bus.getNum(), AcEquationType.BUS_V);
+            if (creationParameters.isReactivePowerControl()) {
```

Comment:
Same concern with this, it it kind of hack that is done after createVoltageControlEquations. We could have a single place to configure control coming from generators.

Reply:
I'm not sure I understood properly what you asked @geofjamg: I put this reactive power control code in a `createGeneratorControlEquations` method, together with the content of `createVoltageControlEquations`, using the fact that it is either voltage control or reactive power control